lib/gcov/include/ukgcov.h

```diff
@@ -0,0 +1,59 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+#ifndef __UKGCOV_H__
+#define __UKGCOV_H__
+
+static const size_t INITIAL_BUFFER_SIZE = 100 * (1 << 10); // 100 KB
+
+/* The start and end symbols are provided by the linker script.  We use the
+   array notation to avoid issues with a potential small-data area.  */
+extern const struct gcov_info *const __gcov_info_start[];
+extern const struct gcov_info *const __gcov_info_end[];
+
+/* Set all counters to zero.  */
+
+extern void __gcov_reset (void);
+
+/* Write profile information to a file.  */
+
+extern void __gcov_dump (void);
+
+/* Convert the gcov information referenced by INFO to a gcda data stream.
```

Comment:
⚠️ **Careful here!** This is from https://github.com/gcc-mirror/gcc/blob/master/libgcc/gcov.h, which **is licensed under the GPL** 💀

Moreover, only `dump_gcov_info()` needs to be public. The rest of the definitions can be moved to `ukgcov.c`. The documentation is not necessary, you can add a comment with a link to https://github.com/gcc-mirror/gcc/blob/master/libgcc/gcov.h, if you think it's necessary.

According to Unikraft's coding convention's, the public headers need to be name-spaced, so please create a `uk` subdirectory under `include/`, ie `include/uk/`. And since this header is now name-spaced, you can now call the header `gcov.h` (without the `uk` prefix). Your users will then do:
```
#include <uk/gcov.h>
```

Finally, I would also prefix the exported symbol with `gcov`, ie rename `dump_gcov_info()` to `gcov_dump_info()`.